bundles/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/PowerMaxGenericBindingProvider.java

```diff
@@ -0,0 +1,140 @@
+                case EVENT_LOG:
+                case PARTITION_STATUS:
+                case PARTITION_ARM_MODE:
+                case COMMAND:
+                    if (!(item instanceof StringItem)) {
+                        throw new BindingConfigParseException(
+                                "item '" + item.getName() + "' is of type '" + item.getClass().getSimpleName()
+                                        + "', only String is allowed - please check your configuration");
+                    }
+                    break;
+                case ZONE_STATUS:
+                    if (!(item instanceof SwitchItem || item instanceof ContactItem)) {
+                        throw new BindingConfigParseException(
+                                "item '" + item.getName() + "' is of type '" + item.getClass().getSimpleName()
+                                        + "', only Switch and Contact are allowed - please check your configuration");
+                    }
+                    break;
+                case ZONE_ARMED:
+                case X10_STATUS:
+                    if (!(item instanceof SwitchItem || item instanceof StringItem)) {
+                        throw new BindingConfigParseException(
+                                "item '" + item.getName() + "' is of type '" + item.getClass().getSimpleName()
+                                        + "', only Switch and String are allowed - please check your configuration");
+                    }
+                    break;
+                case ZONE_LAST_TRIP:
+                    if (!(item instanceof DateTimeItem)) {
+                        throw new BindingConfigParseException(
+                                "item '" + item.getName() + "' is of type '" + item.getClass().getSimpleName()
+                                        + "', only DateTime is allowed - please check your configuration");
+                    }
+                    break;
+                case PARTITION_READY:
+                case PARTITION_BYPASS:
+                case PARTITION_ALARM:
+                case PANEL_TROUBLE:
+                case PANEL_ALERT_IN_MEMORY:
+                case PARTITION_ARMED:
+                case ZONE_BYPASSED:
+                case ZONE_LOW_BATTERY:
+                case PGM_STATUS:
+                    if (!(item instanceof SwitchItem)) {
+                        throw new BindingConfigParseException(
+                                "item '" + item.getName() + "' is of type '" + item.getClass().getSimpleName()
+                                        + "', only Switch is allowed - please check your configuration");
+                    }
+                    break;
+                default:
+                    break;
+            }
+        } catch (IllegalArgumentException e) {
+            throw new BindingConfigParseException(
+                    "item '" + item.getName() + "': parse error of string " + bindingConfig);
+        }
+    }
```

Comment:
Othet bindings are doing that.
I think it is a good way to inform the user of inappropriate item types.
I could suppress these checks and the user will have items never set without any hints.